clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheActiveEntity.java

```diff
@@ -401,8 +401,11 @@ private void validateClientConnected(ClientDescriptor clientDescriptor) throws C
   }
 
   private void validateClientAttached(ClientDescriptor clientDescriptor) throws ClusterException {
-    validateClientConnected(clientDescriptor);
-    if (!clientStateMap.get(clientDescriptor).isAttached()) {
+    ClientState clientState = this.clientStateMap.get(clientDescriptor);
```

Comment:
I would rather have another internal method that does what's in lines 398 to 400 and back in 406 to 407 but indeed takes the `ClientState` instead of looking it up - and being racy indeed.